crypto/camellia/cmll_local.h

```diff
@@ -22,8 +22,8 @@
  * to the OpenSSL project.
  */
 
-#ifndef HEADER_CAMELLIA_LOCL_H
-# define HEADER_CAMELLIA_LOCL_H
+#ifndef OSSL_CRYPTO_CAMELLIA_CMLL_LOCAL_H
+# define OSSL_CRYPTO_CAMELLIA_CMLL_LOCAL_H
```

Comment:
TODO: shorten to `OSSL_CRYPTO_CAMELLIA_LOCAL_H`

_(actually the basename should be fixed, too: `camellia_local.h`)_

Reply:
Why does it even have a guard?  Most of our local headers don't...

Reply:
I‘ll check that.

Reply:
The logic is that thee local headers are only interesting to .c files in the same directory (and some of the internal test programs in `test`), and should never be included by any other .h file.  Therefore, no guards should be generally needed.  There may be exceptions to this line of thinking, but I cannot see how the camellia one would be one of those...

Reply:
Hmmm... there are _a lot_ of local header files with include guards...

https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L10
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L13
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L14
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L16
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L17
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L18
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L19
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L20
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L23
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L29
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L30
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L33
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L34
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L35
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L37
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L39
https://github.com/openssl/openssl/blob/2909f93158d798956648259c702ef2259b1deb36/logfiles/fix-include-guards.sed#L201

Reply:
I think I'll leave this task for a separate pr. Otherwise this one will never get merged...

Reply:
> local headers are only interesting to .c files in the same directory (and some of the internal test programs in `test`), and should never be included by any other .h file.

This seems to be true indeed. The following search yields no results (on this pr's branch).

```make
find -name '*.h*' -exec grep -nH '# *include.*_local\.h' {} \;
```

Reply:
> I think I'll leave this task for a separate pr.

Seventeen local header files have include guards which can be removed. Since the #indentation of the other preprocessor directives in the file needs to be adjusted accordingly, it is simpler to do it manually (with emacs' help) in a separate pull request after this pull request and #9681 is merged.

```
~/src/openssl$ git grep -l _LOCAL_H | grep -v logfiles
crypto/aes/aes_local.h
crypto/bf/bf_local.h
crypto/bn/bn_local.h
crypto/camellia/cmll_local.h
crypto/cmp/cmp_local.h
crypto/cms/cms_local.h
crypto/crmf/crmf_local.h
crypto/des/des_local.h
crypto/ec/curve448/curve448_local.h
crypto/engine/eng_local.h
crypto/hmac/hmac_local.h
crypto/rand/rand_local.h
crypto/rc4/rc4_local.h
crypto/rsa/rsa_local.h
crypto/seed/seed_local.h
crypto/ui/ui_local.h
ssl/ssl_local.h
```

Reply:
> TODO: shorten to `OSSL_CRYPTO_CAMELLIA_LOCAL_H`

Consequently, I won't bother to change the include guard name.